### PR TITLE
Updates setup dependencies and iotrace cli syntax

### PR DESCRIPTION
**Pull request by mariuszbarczak**

### CODE REVIEW

File: setup_dependencies.sh [hunk 1]
```diff
@@ -56,103 +25,144 @@ function detect_distribution ()
...
+}
 
-    setup_other_deps
-    ;;
-"UBUNTU18")
-    info "Ubuntu 18 detected"
-    packages="linux-headers-$(uname -r)"
+if [ "$EUID" -ne 0 ]
+then
+    iotrace_error "Please run as root to allow using pakcage manager"
```
Comment (mmkayPL) [inline setup_dependencies.sh]: package

File: setup_dependencies.sh [hunk 2]
```diff
@@ -56,103 +25,144 @@ function detect_distribution ()
...
-"RHEL7")
-    info "RHEL7.x detected"
-    packages="kernel-devel-$(uname -r)"
+function iotrace_get_kernel_package () {
+    distro=$(detect_distribution)
+    case "${distro}" in
+    "RHEL7"|"CENTOS7")
+        echo "kernel-devel"
+        ;;
+    "FEDORA")
```
Comment (tomaszrybicki) [inline setup_dependencies.sh]: Add to switch above?

File: setup_dependencies.sh [hunk 3]
```diff
@@ -56,103 +25,144 @@ function detect_distribution ()
...
+    *)
+        iotrace_error "Unknown Linux distribution"
+        ;;
+    esac
+}
 
-    info "Installing packages: ${packages}"
-    yum -y install ${packages}
-    check_result $? "Cannot install required dependencies"
+function iotrace_get_distribution_pkg_depndenices () {
```
Comment (tomaszrybicki) [inline setup_dependencies.sh]: dependencies

File: setup_dependencies.sh [hunk 4]
```diff
@@ -56,103 +25,144 @@ function detect_distribution ()
...
+    ${PKG_INSTALLER} $(iotrace_get_kernel_package)
+    iotrace_check_result $? "Cannot install kernel headers"
 
-    if [ ! -d /usr/src/kernels/$(uname -r) ]
+    kernel_dir=$(readlink -f /lib/modules/${kernel_version}/build)
+    if [ ! -d "${kernel_dir}" ]
     then
-        error "Linux kernel headers were not properly installed."
+        iotrace_info "Please run the kernel appropriate to the kernel headers"
+        iotrace_error "Installed kernel headers do not match to the running kernel version."
```
Comment (tomaszrybicki) [inline setup_dependencies.sh]: Maybe let's add a note "Consider updating system kernel to match available headers" to suggest next action.

File: setup_dependencies.sh [hunk 5]
```diff
@@ -56,103 +25,141 @@ function detect_distribution ()
...
+}
 
-    info "Installing packages: ${packages}"
-    yum -y install ${packages}
-    check_result $? "Cannot install required dependencies"
+function iotrace_get_distribution_pkg_dependencies () {
+    distro=$(detect_distribution)
+    case "${distro}" in
+    "RHEL7"|"CENTOS7"|"FEDORA"|"UBUNTU")
+        echo "rpm-build dpkg"
```
Comment (tomaszrybicki) [inline setup_dependencies.sh]: For ubuntu it is just "rpm"

File: tests/utils/installer.py [hunk 6]
```diff
@@ -103,7 +103,7 @@ def check_if_installed():
 
 def uninstall_iotrace():
     TestRun.LOGGER.info("Uninstalling previous iotrace")
-    TestRun.executor.run_expect_success(
+    TestRun.executor.run(
```
Comment (tomaszrybicki) [inline tests/utils/installer.py]: This step used to fail entire execution when installting iotrace on a fresh system - cd failed.

**Pull request merged**